src/Workspaces/Core/Portable/Diagnostics/HostAnalyzerAssemblyLoaderProvider.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.IO;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Host;
+
+/// <summary>
+/// Similar to <see cref="DefaultAnalyzerAssemblyLoader"/> but uses a path appropriate for VS/VSCode host workspace.
+/// </summary>
+[ExportWorkspaceService(typeof(IAnalyzerAssemblyLoaderProvider), [WorkspaceKind.Host]), Shared]
+internal class HostAnalyzerAssemblyLoaderProvider : IAnalyzerAssemblyLoaderProvider
```

Comment:
I'm not sure if we want to move this to the language server/VS layers, since this might mess with other hosts just to be safe.

Reply:
split into vs and vscode version, moved vs version into editorfeatures.